forge-game/src/main/java/forge/game/ability/effects/ManaEffect.java

```diff
@@ -240,7 +241,8 @@ else if (cs.isMonoColor())
                 continue;
             }
 
-            abMana.produceMana(mana, p, sa);
+            abMana.produceMana(mana, p, sa, runTriggers);
+            runTriggers = false; //avoid running TapsForMana triggers for each player
```

Comment:
The Taps for Mana Trigger should be outside of this loop

Reply:
The trigger is dependent on `ProduceMana` RE though

if we keep that inside `produceMana` needs to return replaced mana Strings that are all combined?

Reply:
i would need to checkout the trigger and existing, but TapsForMana probably need to get the ProducedMana as Player => String Map?

we need to check if the TapsForMana trigger care about who is tapping for mana (probably you) and for who the mana is produced. (i don't think that current trigger does care for this)

Reply:
Yea I see no trigger where it matters